system-contracts/contracts/L1Messenger.sol

```diff
@@ -259,7 +260,8 @@ contract L1Messenger is IL1Messenger, SystemContractBase {
         // Shift calldata ptr past the pubdata offset and len
         calldataPtr += 64;
 
-        /// Check logs
+        /// Check logs. Full logs root hash is published even for Validiums without DA.
+        // TODO: makes sense to move to L2DAValidator as part of pubdata validation, but idk if we want to inflate audit scope
```

Comment:
I think it may be hard to move it, we also need to verify/hash logs for the l2 logs tree as you already wrote.

But IMHO it may make sense to move validation of other fields(bytecodes, messages, state diffs) in this method, but we'll have to ignore it somehow for validiums (no da), which may not be beautiful.

Reply:
It's not hard, since L2DAValidator is a library now, we can do it and move all validation there. It's just that we can keep it as is and reduce diff for the audit. In BoojumOS we can regroup logic as we want

Reply:
I mean it will be hard if you don't want to reduce performance, because, in fact, you need to build an l2 logs tree here. So with the current `operatorInput` you'll have to hash logs here and later during the validation.

Reply:
> you'll have to hash logs here
But we could move it completely to that lib and just get root hash as an output